src/sync.rs

```diff
@@ -42,6 +45,17 @@ where
     IO: DfuIo<Read = usize, Write = usize, Reset = (), Error = E>,
     E: From<std::io::Error> + From<Error>,
 {
+    /// Download a slice to on to the device
+    pub fn download_from_slice(&mut self, slice: &[u8]) -> Result<(), IO::Error> {
+        let length = slice.len();
+        let cursor = Cursor::new(slice);
+
+        self.download(
+            cursor,
+            u32::try_from(length).map_err(|_| std::io::Error::from(ErrorKind::InvalidData))?,
```

Comment:
Oh sorry I just realize now it's not an std::io::Error but an IO::Error that this function returns. So you can just add a variant in in the Error enum in lib.rs. Something like "The size of the data being transferred exceeds the DFU capabilities.".

(Don't forget the dots at the end of lines in doc comments. It seems quite standard in Rust libs)